### ISSUE TITLE
Pressure difference in use case UCPipeS01

### ISSUE THREAD

**Issue by marcusfuchs** [bramvdh91/modelica-annex60#24]
From the use case description: 

> This use case aims at demonstrating the pressure loss behavior of the pipe model. In theory, two pipe models of length 50 m in series should behave the same as one pipe model with length of 100 m. To demonstrate this behavior, the two 50 m pipe models are placed in parallel to one 100 m pipe between a source and a sink model. The pressure difference between source and sink is varied following a sine function. As temperature is not relevant for this use case, it is kept constant.

Currently, there is a difference between the two 50 m pipes and the 100 m pipe as shown in `Annex60.Experimental.Pipe.Examples.UseCases.TypeS_Development.UCPipeS01AD_Friction` (currently in branch `issue264_pipeUseCases` but #22 should merge this into `issue264_pipe`). Here is a part of the plot illustrating the difference:

![ucs1](https://cloud.githubusercontent.com/assets/5516900/15962054/c092667c-2f08-11e6-91cf-4ea2476bc388.png)

I remember seeing this behavior a while back, and thought we had already fixed that, but I'll look into it again.

**Comment by bramvdh91** [comment]
@marcusfuchs , do you recall where this issue has been treated before and what was done to solve it? 

_edit:_ created branch `pipe_issue24_dp` for this issue.

**Comment by bramvdh91** [comment]
It seems that the difference starts as soon as the flow goes from turbulent to laminar (see image below, where m_flow_turbulent has been plotted as well).
![ucpipes01ad_friction](https://cloud.githubusercontent.com/assets/10756543/16587205/caf2a9fe-42c9-11e6-9dd3-cefcecc0b22f.png)

My guess is that the erroneous behaviour results from the homotopy that's being used to end up with a twice differentiable mass flow rate. 

It seems that the difference disappears when `use_dh` is set to `false`.

**Comment by mwetter** [comment]
@marcusfuchs , @bramvdh91 

The model `UCPipeS01AD_Friction` does not exist on `pipe_issue24_dp` so I cannot reproduce it, but it looks like the reason is that the transition to the regularized region occurs at

```
m_flow_turbulent = if (computeFlowResistance and use_dh) then 
                       eta_default*dh/4*Modelica.Constants.pi*ReC
                       elseif (computeFlowResistance) then 
                       deltaM * m_flow_nominal_pos
```

Hence, different computations are used depending on the value of `use_dh`.

As this causes confusion, I wonder if `FixedResistanceDpM` should be changed so that `use_dh, dh, ReC` are removed and the model is renamed to `FixedResistance_dp`, and another model `FixedResistance_dh` is created. Renaming the model allows to move the original model to `Annex60.Obsolete.Fluid.FixedResistances` and use of a conversion script so that old models still work for a while until the model is removed.

This change should however be done on a new branch that forks of the `master` so that it can be done cleanly and then merged into your development branch(es).

@bramvdh91 Note that the homotopy is a different concept than the use of twice differentiable functions. Homotopy is used to replace an equation with a simpler one in case of convergence problems during initialization. However when the equation of the homotopy converged, the solver transitions to the original model before the time stepping starts.

**Comment by bramvdh91** [comment]
> The model `UCPipeS01AD_Friction` does not exist on `pipe_issue24_dp` so I cannot reproduce it, but it looks like the reason is that the transition to the regularized region occurs at
> 
> ```
> m_flow_turbulent = if (computeFlowResistance and use_dh) then 
>                        eta_default*dh/4*Modelica.Constants.pi*ReC
>                        elseif (computeFlowResistance) then 
>                        deltaM * m_flow_nominal_pos
> ```
> 
> Hence, different computations are used depending on the value of use_dh.

Strange that `UCPipeS01AD_Friction` did not appear, I've checked and I found it [here](https://github.com/bramvdh91/modelica-annex60/blob/pipe_issue24_dp/Annex60/Experimental/Pipe/Examples/UseCases/TypeS_Development/UCPipeS01AD_Friction.mo).
I also tried to track down the difference. However, when I change `m_flow_nominal` such that `m_flow_turbulent` is the same for both values of `use_dh`, the difference between the series and single mass flow does not appear. 

I can split up `FixedResistanceDpM`, however I'm not familiar with the conversion script. I'll start doing this in `bram_issue24_fixedRes`, branching from the current Annex 60 master. 
- [x] Write new classes 
- [x] Adapt documentation
- [ ] Add conversion script
- [ ] Update examples in `Annex60.Fluid.FixedResistances.Examples`

**Comment by bramvdh91** [comment]
I have added an example to compare both versions of the fixedResistance component. 
I chose a pipe of 10 m of DN100 with a nominal pressure drop of 100 Pa/m and a nominal mass flow rate of about 8 kg/s. I found that the mass flow rate where this pipe transitions to turbulent flow is around 0.005 kg/s. If this were to be modelled with a `FixedResistance_dp`, DeltaM would have to be much lower than the current standard of 0.3 (and minimum value 0.01, which is still too high).
For this example, the difference between the mass flow through two pipes of 5 m or one pipe of 10 m is negligible (order 1E-11).

**Comment by bramvdh91** [comment]
As you can see in the latest commit on `pipe_issue24_dp` the difference in mass flow is solved by setting `from_dp=true` for this case.

**Comment by bramvdh91** [comment]
After a discussion with @mathadon , I implemented another version where the current version of `FixedResistanceDpM` can be kept as it is, except for deleting `use_dh` and `dh`. `FixedResistanceDhM` extends from this class.
However, due to the reassignment of `m_flow_turbulent` in the latter, this parameter can no longer be set `final` in `...DpM`. In order to keep it out of sight, I have moved it to the "Advanced" tab. Furthermore, `deltaM` is not used in `FixedResistanceDhM` and hence set to an arbitrary value and declared `final`.
The reason for this version is to keep the number of added classes low (since the extra base class in the first implementation does not help readability of the code). 
This version is located in a different branch, `bram_issue24_fixedResLessPartials`.

@mwetter , could you review these versions and tell me which one you would prefer? If the first version (in `bram_issue24_fixedRes`) is chosen, conversion scripts need to be added, or the old version of `FixedResistanceDpM` could of course be replaced.

**Comment by bramvdh91** [comment]
Closing this issue as we are now using ``HydraulicDiameter``.
